core/src/main/scala/com/eed3si9n/jarjarabrams/Shader.scala

```diff
@@ -25,9 +25,17 @@ object Shader {
 
   def shadeInputStreams(
       rules: Seq[ShadeRule],
-      mappings: Seq[(InputStream, String)],
+      mappings: Iterable[(InputStream, String)],
       verbose: Boolean
-  ): Seq[(InputStream, String)] = {
+  ): Iterable[(InputStream, String)] = {
+    val shader = inputStreamShader(rules, verbose)
+    mappings.flatMap(f => shader(f._1, f._2))
+  }
+
+  def inputStreamShader(
+    rules: Seq[ShadeRule],
+    verbose: Boolean
+  ): (InputStream, String) => Option[(InputStream, String)] = {
```

Comment:
Wouldn't `(Array[Byte], String) => Option[(Array[Byte], String)]` work? As the `InputStream`s are converted to / from `Array[Byte]` / `ByteArrayInputStream`, we could just pass `Array[Byte]`s around instead.

Reply:
I pushed a version with that